docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,804 @@
+
+For these new implementations, the current code for `File`, `SocketClient`, and `RetrySocketClient` will have to implement the `CryptoProvider` interface instead of the `PrivValidator` one.
+
+##### Code snippet for `CryptoProviderPV`
+
+As mentioned above, instead of having several implementations of `PrivValidator`, the proposal is to have only one that, by dependency injection, loads the corresponding `CryptoProvider` that offers the same functionality as the previous implementations of `PrivValidator`.
+
+Below is an example of how `CryptoProviderPV` would look like. Note that in this particular case, since the PrivateKey is managed inside the corresponding implementation, we're not passing that value to the signer. This is to avoid having to significantly change the code for `FilePV`. This is also valid for all implementations that manage their private keys in their own logic.
+
+```go
+// CryptoProviderPV is the implementation of PrivValidator using CryptoProvider's methods
+type CryptoProviderPV struct {
+    provider CryptoProvider
+    pubKey   PubKey
+}
+
+// NewCryptoProviderPV creates a new instance of CryptoProviderPV
+func NewCryptoProviderPV(provider CryptoProvider, pk PubKey) (*CryptoProviderPV, error) {
+    return &CryptoProviderPV{provider: provider, pubKey: pubKey}, nil
+}
+
+// GetPubKey returns the public key of the validator
+func (pv *CryptoProviderPV) GetPubKey() (PubKey, error) {
+    return pv.pubKey, nil
+}
+
+// SignVote signs a canonical representation of the vote. If signExtension is true, it also signs the vote extension.
+func (pv *CryptoProviderPV) SignVote(chainID string, vote *Vote, signExtension bool) error {
+    signer := pv.provider.GetSigner()
+
+    // code for getting voteBytes goes here
+    // voteBytes := ...
+
+    // The underlying signer needs these parameters so we pass them through SignerOptions
+    options := SignerOptions{
+        "chainID": chainID,
+        "vote":    vote,
+    }
+ 
+    sig, _ := signer.Sign(voteBytes, options)
+    vote.Signature = sig
+    return nil
+}
+
+// SignProposal signs a canonical representation of the proposal
+func (pv *CryptoProviderPV) SignProposal(chainID string, proposal *Proposal) error {
+    signer := pv.provider.GetSigner()
+
+    // code for getting proposalBytes goes here
+    // proposalBytes := ...
+
+    // The underlying signer needs these parameters so we pass them through SignerOptions
+    options := SignerOptions{
+        "chainID":  chainID,
+        "proposal": proposal,
+    }
+ 
+    sig, _ := signer.Sign(proposalBytes, options)
+    proposal.Signature = sig
+    return nil
+}
+
+// SignBytes signs an arbitrary array of bytes
+func (pv *CryptoProviderPV) SignBytes(bytes []byte) ([]byte, error) {
+    signer := pv.provider.GetSigner()
+    return signer.Sign(bytes, SignerOptions{})
+}
+
+```
+
+*Note:* Each provider (File, Socket, SignerClient, RetrySignerClient) will need to be modified to satisfy the `CryptoProvider` interface.
+
```

Comment:
We need to add some text with an analysis on the changes required to TMKMS (is any required?)

Reply:
ok, I'm not aware of that code, could please point me to the files that I should look?